### PR TITLE
Config: Autoscan list to plain vector

### PR DESCRIPTION
**Pull request by whyman**

### CODE REVIEW

**Comment by whyman** [comment]
Im not really sure it makes sense to allow editing the autoscan items from config via the config UI.

If you want to edit them via the UI just do it via the usual method...

1st stage of moving out autoscan management out of content manager.

File: src/config/config_setup.h [hunk 1]
```diff
@@ -662,9 +662,9 @@ class ConfigAutoscanSetup : public ConfigSetup {
     /// \brief Creates an array of AutoscanDirectory objects from a XML nodeset.
     /// \param element starting element of the nodeset.
     /// \param scanmode add only directories with the specified scanmode to the array
-    bool createOptionFromNode(const pugi::xml_node& element, std::shared_ptr<AutoscanList>& result);
+    bool createOptionFromNode(const pugi::xml_node& element, std::vector<AutoscanDirectory> result);
```
Comment (KarlStraussberger) [inline src/config/config_setup.h:665]: You need a reference otherwise no result is returned

Comment (whyman) [reply src/config/config_setup.h:665]: Good spot

File: src/config/config_options.h [hunk 2]
```diff
@@ -233,15 +234,15 @@ class ArrayOption : public ConfigOption {
 
 class AutoscanListOption : public ConfigOption {
 public:
-    explicit AutoscanListOption(std::shared_ptr<AutoscanList> option)
-        : option(std::move(option))
+    explicit AutoscanListOption(std::vector<AutoscanDirectory> autoscans_)
+        : autoscans(std::move(autoscans_))
```
Comment (neheb) [inline src/config/config_options.h:238]: shouldn't this be opposite? WebKit style says _ is for member variables AFAIK.

File: src/config/config_setup.h (hunk 1 above)
Comment (whyman) [reply src/config/config_setup.h:665]: Yep

File: src/config/config.h [hunk 3]
```diff
@@ -30,8 +30,8 @@
 #include "util/grb_fs.h"
 
 // forward declarations
-class AutoscanDirectory;
 class AutoscanList;
```
Comment (KarlStraussberger) [inline src/config/config.h:33]: Isn't this obsolete now?

**Comment by KarlStraussberger** [comment]
> Im not really sure it makes sense to allow editing the autoscan items from config via the config UI.

We should keep the option to check the settings, but normally this is done in config.xml

**Comment by whyman** [comment]
Yeah, I didnt remove it, was more a general comment.

**Comment by KarlStraussberger** [comment]
Ready to merge?

**Comment by whyman** [comment]
I think so!

There isnt great test coverage for this, but I've run it locally and done some things and it seemed to work as I would expect.

**Pull request merged**